labs/src/main/scala/org/scalalabs/basic/lab03/OptionExercise.scala

```diff
@@ -25,9 +25,14 @@ object OptionExercise01 {
    * - does not exist: 					"not existing"
    */
   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
+          if (room > rooms.size) "not existing" 
+          else{
+            val workingRoom = rooms.get(room)
```

Comment:
I think you might miss an edge case the rooms are not necessarily defined in ascending numerical order - for instance, if only rooms (1,2,4) exist and you search for room 3, you'll make it past the conditional and then crash when you try to use the value of workingRoom (which will be null).